api/views.py

```diff
@@ -1,18 +1,19 @@
 from django.http import JsonResponse
 from django.contrib.auth.decorators import login_required
-
-from oauth2_provider.views import ScopedProtectedResourceView
+from oauth2_provider.views import TokenView
 
 from core.decorators import class_decorator
 
 
 @class_decorator(login_required)
-class ProfileView(ScopedProtectedResourceView):
-    required_scopes = ['read']
+class ProfileView(TokenView):
 
     def get(self, request, *args, **kwargs):
         user = request.user
-        return JsonResponse({
-            'id': user.pk,
-            'email': user.email,
-        })
+        _, request = self.verify_request(request)
+        response = {'id': user.pk}
+
+        if request.access_token.scope == 'email':
```

Comment:
We will probably need another _scope_ for the groups authorities for datagouv.
However I'm not sure it will behave the same way and I wouldn't go for abstration while we don't need it.